clientconn.go

```diff
@@ -159,23 +160,20 @@ func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *
 		}
 	}()
 
-	if channelz.IsOn() {
-		if cc.dopts.channelzParentID != 0 {
-			cc.channelzID = channelz.RegisterChannel(&channelzChannel{cc}, cc.dopts.channelzParentID, target)
-			channelz.AddTraceEvent(logger, cc.channelzID, 0, &channelz.TraceEventDesc{
-				Desc:     "Channel Created",
-				Severity: channelz.CtInfo,
-				Parent: &channelz.TraceEventDesc{
-					Desc:     fmt.Sprintf("Nested Channel(id:%d) created", cc.channelzID),
-					Severity: channelz.CtInfo,
-				},
-			})
-		} else {
-			cc.channelzID = channelz.RegisterChannel(&channelzChannel{cc}, 0, target)
-			channelz.Info(logger, cc.channelzID, "Channel Created")
+	pid := cc.dopts.channelzParentID
+	cc.channelzID = channelz.RegisterChannel(&channelzChannel{cc}, pid, target)
```

Comment:
If channelz is not turned on, those IDs will all be nil, and then the logs will all have `[channel(id:0)]`, right?

Reply:
Actually, they all have `[]` empty string
```
    tlogger.go:116: INFO clientconn.go:106 [core] Channel(id:0) created  (t=+498.514µs)
    tlogger.go:116: INFO clientconn.go:1624 [core] [] original dial target is: "whatever:///test.server"  (t=+559.064µs)
```

Reply:
Maybe we can generate channelz IDs even if channelz is off. It's just a self incrementing int anyway.

Reply:
Made the change to generate channelz IDs even when channelz is OFF. This also required changes to some tests. I've captured these changes as a separate commit. https://github.com/grpc/grpc-go/pull/5192/commits/786e6be418c495da5a42f3490f5a007d9acc5e31

Reply:
Did you try deleting the `==nil` checks in channelz ID's methods?
Should all channelz IDs be non-nil now? Maybe not true in tests. But still might be interesting to try.